client/state/sharing/keyring/reducer.js

```diff
@@ -28,11 +28,11 @@ export const isFetching = createReducer( false, {
 // Stores the list of available keyring connections
 export const items = createReducer( {}, {
 	[ KEYRING_CONNECTIONS_RECEIVE ]: ( state, { connections } ) => ( { ...keyBy( connections, 'ID' ) } ),
-	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: [ ...state[ ID ].sites, siteId.toString() ]
+	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { connection: { keyring_connection_ID: ID, site_ID } } ) => ( { ...state, [ ID ]: {
+		...state[ ID ], sites: [ ...state[ ID ].sites, site_ID.toString() ]
```

Comment:
I also think we should be very wary about destructuring more than a single level deep, as it becomes very difficult to read, especially with aliasing. Speaking of, naming should be `id`, not `ID`, per [camelCasing guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#naming-conventions).